### PR TITLE
[fix8] Buffer overflow protection

### PR DESCRIPTION
**Pull request by earce**
Current FIX8 behavior creates a segmentation fault if a message being encoded exceeded the default library buffer size (8k). 

- This change guards from writing beyond the region memory owned by the buffer
- It will also throw a `BufferBoundaryExceeded` which the caller can catch if the buffer capacity is exceeded
- Adds unit test to validate behavior
- Heavily used https://github.com/IG-Group/fix8/pull/19 as inspiration

### CODE REVIEW

**Comment by Molly-Brown** [comment]
Can one of the admins verify this patch?

File: include/fix8/message.hpp [hunk 1]
```diff
@@ -883,10 +885,6 @@ class MessageBase
 	   \return reference to FieldTraits object */
 	const FieldTraits& get_fp() const { return _fp; }
 
-	/*! Get the FieldTraits
-		\return mutable reference to FieldTraits object */
-	FieldTraits& mutable_fp() const { return _fp; }
```
Comment (naveensanmane) [inline include/fix8/message.hpp:888]: this function was recently added. any reason for removal ?

File: runtime/message.cpp [hunk 2]
```diff
@@ -359,14 +359,21 @@ size_t MessageBase::encode(char *to) const
 		Presence::const_iterator fpitr(_fp.get_presence().end());
 		if (!_fp.get(pp.second->_fnum, fpitr, FieldTrait::suppress))	// some fields are not encoded until unsuppressed (eg. checksum)
 		{
+            if (to > buf_boundary) {
```
Comment (naveensanmane) [inline runtime/message.cpp:362]: indentation seems off.

File: include/fix8/message.hpp (hunk 1 above)
Comment (earce) [reply include/fix8/message.hpp:888]: I am having build issues with this locally

Comment (naveensanmane) [reply include/fix8/message.hpp:888]: ok sure. not used yet. we can address this later

**Pull request merged**